nofos/nofos/nofo.py

```diff
@@ -834,6 +842,74 @@ def get_cover_image(nofo):
     return "img/cover.jpg"
 
 
+def upload_cover_image_to_s3(nofo, uploaded_file, alt_text=""):
+    """
+    Uploads a cover image to S3 and updates the NOFO.
+
+    Args:
+        nofo: The NOFO object to update
+        uploaded_file: The uploaded file from request.FILES
+        alt_text: Optional alt text for the image
+
+    Returns:
+        bool: True if upload and update succeed.
+
+    Raises:
+        ValidationError: If file validation fails for the following reasons:
+        - File size exceeds 5MB
+        - File type is not allowed
+        - File extension is not allowed
+    """
+    # Validate file size (5MB limit)
+    max_size = 5 * 1024 * 1024  # 5MB in bytes
+    if uploaded_file.size > max_size:
+        raise ValidationError(
+            f"File size ({uploaded_file.size / (1024*1024):.1f}MB) exceeds maximum allowed size (5MB)."
+        )
+
+    # Validate file type
+    allowed_types = ["image/png", "image/jpg", "image/jpeg"]
+    if uploaded_file.content_type not in allowed_types:
+        raise ValidationError(
+            f"Invalid file type: {uploaded_file.content_type}. Please upload a valid image file."
+        )
+
+    # Validate file extension
+    allowed_extensions = [".png", ".jpg", ".jpeg"]
+    file_extension = "." + uploaded_file.name.lower().split(".")[-1]
+    if file_extension not in allowed_extensions:
+        raise ValidationError(
+            f"Invalid file extension: {file_extension}. Allowed extensions: {', '.join(allowed_extensions)}"
+        )
+
+    # Rename file to have NOFO name
+    uploaded_file.name = f"{nofo.number}{file_extension}".lower()
```

Comment:
Technically it is allowed for NOFOs to have no number (although very unlikely). However, a NOFO must have either a number _or_ a title. So let's say "nofo.number" or "nofo.title".replace(" ", "-") maybe?

Reply:
Otherwise you can do this

<img width="364" height="159" alt="Screenshot 2025-08-01 at 10 25 41" src="https://github.com/user-attachments/assets/caacafb3-ce2d-440b-a48d-c3a57e364d0e" />